http4s/client/src/main/scala/endpoints4s/http4s/client/Endpoints.scala

```diff
@@ -239,22 +239,29 @@ trait EndpointsWithCustomErrors
         .map(_.andThen(_.map(_.asLeft[B])))
         .orElse(responseB(sc, hs).map(_.andThen(_.map(_.asRight[A]))))
 
-  //#endpoint-type
-  type Endpoint[A, B] = Kleisli[Effect, A, B]
-  //#endpoint-type
+  trait Endpoint[A, B] {
+    def send(request: A): Resource[Effect, B]
+
+    /** Note that this method CAN be unsafe if a lazy resource is returned as the Response due to closing the underlying Resource
+      *
+      * @param request
+      * @return
+      */
+    def sendAndConsume(request: A): Effect[B] = send(request).use(effect.pure)
```

Comment:
Maybe consider adding a deprecated `apply` method that would suggest using `sendAndConsume` instead, to ease migration?